### PR TITLE
core/scripts/common: rm ava-labs/coreth

### PR DESCRIPTION
**Pull request by jmank88**
This is an experiment to see if we can shed the `ava-labs/coreth` dependency in favor of using an identical custom type.

### CODE REVIEW

File: core/scripts/common/vrf/setup-envs/main.go [hunk 1]
```diff
@@ -211,6 +212,7 @@ func main() {
 				coordinatorConfigV2,
 				*batchFulfillmentEnabled,
 				nodesMap,
+				true,
```
Comment (jmank88) [inline core/scripts/common/vrf/setup-envs/main.go:215]: This was already broken separately from a signature change. I'm not sure that true is the correct value.

Comment (iljapavlovs) [reply core/scripts/common/vrf/setup-envs/main.go:215]: latest develop should have it fixed already, so better sync with develop

Comment (jmank88) [reply core/scripts/common/vrf/setup-envs/main.go]: Updated!

File: core/scripts/common/avalance.go
Comment (makramkd) [inline core/scripts/common/avalance.go:1]: File name: `avalanche.go`?

File: core/scripts/common/avalance.go [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+	// correctly encode this field optionally with uint64.
+	ExtDataGasUsed *big.Int `json:"extDataGasUsed" rlp:"optional"`
+
+	// BlockGasCost was added by Apricot Phase 4 and is ignored in legacy
+	// headers.
+	BlockGasCost *big.Int `json:"blockGasCost" rlp:"optional"`
+}
+
+func (h *AvaHeader) UnmarshalJSON(input []byte) error {
+	type Header struct {
```
Comment (makramkd) [inline core/scripts/common/avalance.go:51]: Why the re-decl here? Or is this how the avalanche code looks like?

**Review by makramkd — COMMENTED** [review]
This should work fine in theory, just need to make sure we keep up with avalanche's header updates

**Review by makramkd — APPROVED** [review]

**Pull request merged**